client/lib/plugins/utils.js

```diff
@@ -424,3 +426,25 @@ export function getProductSlugByPeriodVariation( periodVariation, productsList )
 	return Object.values( productsList ).find( ( product ) => product.product_id === productId )
 		?.product_slug;
 }
+
+/**
+ * @param  {object} plugin The plugin object
+ * @param  {boolean} isMarketplaceProduct Is this part of WP.com Marketplace or WP.org
+ * @returns {string} The software slug string
+ */
+export const getSoftwareSlug = ( plugin, isMarketplaceProduct ) =>
+	isMarketplaceProduct ? plugin.software_slug || plugin.org_slug : plugin.slug;
+
+/**
+ * @param  {object} plugin The plugin object
+ * @param  {Array} purchases An array of site purchases
+ * @param  {boolean} isMarketplaceProduct Is this part of WP.com Marketplace or WP.org
+ * @returns {string} The software slug string
+ */
+export const isPluginPurchased = ( plugin, purchases, isMarketplaceProduct ) =>
+	isMarketplaceProduct &&
+	purchases.find( ( purchase ) =>
+		Object.values( plugin?.variations ).some(
```

Comment:
Can `plugin?.variations` be undefined sometimes? On http://calypso.localhost:3000/plugins/browse/paid/example.wpcomstaging.com, sometimes when I click on a plugin to go to the plugin details page, I get a "We're sorry, but an unexpected error has occurred" error page. It's a bit inconsistent (not always the same plugin), but stepping through the browser debugger, it looks like `plugin?.variations` is sometimes undefined?


https://user-images.githubusercontent.com/140841/195126006-e57262ad-0e41-4eac-80f4-e0511a6d91e3.mp4

Reply:
This should be addressed here https://github.com/Automattic/wp-calypso/pull/68602/commits/2c15ccf1f35b91c65fd96639cf5ba17f5e1f6489

Reply:
I'm still seeing the same behavior. Maybe `getPluginPurchased` is trying to return `undefined`?

Reply:
Caught the error 

<img width="838" alt="CleanShot 2022-10-14 at 18 03 25@2x" src="https://user-images.githubusercontent.com/12430020/195879443-047f3d0d-fbc3-4981-b6ae-7a0d6f34cec0.png">

I tried to reproduce it afterwards without luck. Seems being a concurrency issue with the plugin being `null` or `undefined`. This should fix that [c840220](https://github.com/Automattic/wp-calypso/pull/68602/commits/c840220951453e3236964be9db1f2a63e1bd7603). Can you test again?

Reply:
That seems to have done the trick! It's working well for me now! 👍